lib/openstax/aws/secrets.rb

```diff
@@ -140,10 +142,20 @@ def build_secrets(specifications:, substitutions:)
           end
         when /^ssm\((.*)\)$/
           begin
-            client.get_parameter({
-              name: substitutions[$1] || substitutions[$1.to_sym],
+            parameter_name = $1.starts_with?("/") ? $1 : (substitutions[$1] || substitutions[$1.to_sym])
+
+            if parameter_name.blank?
+              raise "#{$1} is neither a literal parameter name nor available in the given substitutions"
+            end
+
+            parameter = client.get_parameter({
+              name: parameter_name,
               with_decryption: true
-            }).parameter.value
+            }).parameter
+
+            type = parameter.type
+
+            parameter.value
           rescue Aws::SSM::Errors::ParameterNotFound => ee
```

Comment:
you could use named captures here instead of global $1 https://ruby-doc.org/core-2.6.5/Regexp.html#method-i-named_captures